lib/phoenix/live_dashboard/pages/metrics_page.ex

```diff
@@ -56,32 +57,30 @@ defmodule Phoenix.LiveDashboard.MetricsPage do
   end
 
   @impl true
-  def render_page(_assigns), do: raise("this page is special cased to use render/2 instead")
-
-  def render(assigns) do
-    ~L"""
-    <div class="row">
-      <div class="container">
-        <ul class="nav nav-tabs mb-4 charts-nav">
-          <%= for group <- @groups do %>
-            <li class="nav-item">
-              <%= live_redirect(format_group_name(group),
-                    to: live_dashboard_path(@socket, :metrics, @page.node, group: group),
-                    class: "nav-link #{if @group == group, do: "active"}") %>
-            </li>
-          <% end %>
-        </ul>
-      </div>
-    </div>
-
-    <%= if @metrics do %>
-      <div class="phx-dashboard-metrics-grid row">
-      <%= for {metric, id} <- @metrics do %>
-        <%= live_component @socket, ChartComponent, id: id, metric: metric %>
+  def render_page(assigns) do
+    tabs =
+      for {name, metrics} <- assigns.metrics_per_tab do
+        {String.to_atom(name),
+         [name: format_tab_name(name), render: render_metrics(Enum.with_index(metrics), assigns)]}
+      end
+
+    tab_bar(tabs: tabs, page: assigns.page)
+  end
+
+  def render_metrics(metrics, assigns) do
+    fn ->
+      assigns = Map.put(assigns, :metrics, metrics)
```

Comment:
We should avoid changing the assigns directly. It is fine to access the metrics below just as `metrics`. :)

Reply:
Another option is to store the metrics we selected on `mount`: https://github.com/phoenixframework/phoenix_live_dashboard/pull/182/files/72dfc8b65380d24517ad18263b5fa1c4dd46d04e..848126f46cda916cb8259bcf9e10a488c793d66c#diff-5f7113279dd1d4426c4943cf8d7cc6c2R18

The truth is: we don't actually need to keep the metrics_per_tab because only one tab is always rendered. So for this page, we just need to store the tab names and the current metrics. The loop code above can become:

      for name <- assigns.tabs do
        {String.to_atom(name), name: format_tab_name(name), render: render_metrics(assigns)}
      end

Reply:
I updated to use just metrics.

About the second comment, I'm not sure what you mean :sweat_smile: . In a previous comment, you agreed with me that the `assigns` params should be in sent to the `render_metrics` from the `render` function, not in `mount`: https://github.com/phoenixframework/phoenix_live_dashboard/pull/182#discussion_r485808436

 In fact, the `mount` function doesn't have `assigns` variable, and the `socket` variable, AFAIK, it is different from the `render` one.

If I misunderstood you, just tell me :)